dwave_networkx/generators/pegasus.py

```diff
@@ -402,6 +402,88 @@ def defragment_tuple(chimera_coords):
     return defragment_tuple
 
 
+def fragmented_edges(pegasus_graph):
+    """
+    Generator for the edges contained in a Chimera graph obtained by splitting each Pegasus node into
+    six Chimera nodes.  If the Pegasus graph has size parameter m, then the derived graph will be a
+    subgraph of Chimera(6m, 6m, 2) -- that is, a Chimera graph with K2,2 unit tiles.
+
+        The K2,2 Chimera graph uses a coordinate system with an origin at the upper left corner of
+        the graph.
+            y: number of vertical fragments from the top-most row
+            x: number of horizontal fragments from the left-most column
+            u: 1 if it belongs to a horizontal qubit, 0 otherwise
+            r: fragment index on the K2,2 shore
+
+    Parameters
+    ----------
+    pegasus_graph: networkx.graph
+        A pegasus graph
+
+    Returns
+    -------
+    (coord0, coord1), ... : an iterator of tuples
+        Yields the edges contained in the Chimera graph derived from the "fragmentation" construction
+    """
+    offsetlist = [pegasus_graph.graph['vertical_offsets'], pegasus_graph.graph['horizontal_offsets']]
+    offsets = {(u, k): offsetlist[u][k] for u in (0, 1) for k in range(12)}
+
+    if pegasus_graph.graph['labels'] == 'nice':
+        coords = pegasus_coordinates.nice_to_pegasus
+    elif pegasus_graph.graph['labels'] == 'int':
+        coords = pegasus_coordinates(pegasus_graph.graph['rows']).linear_to_pegasus
+    else:
+        coords = lambda z: z
+
+    #first, we generate the edges internal to the fragments corresponding to a node
+    for q in pegasus_graph.nodes():
+        u, w, k, z = coords(q)
+        #copied from get_tuple_fragmentation_fn and slightly optimized
+        offset = offsets[u, k]
+        x0 = (z * 12 + offset) // 2
+        y = (w * 12 + k) // 2
+        base = [y, x0, u, k & 1] if u else [x0, y, u, k & 1]
```

Comment:
You're on the right track, but I think you're mistaking the bitwise and, `k&1`, with the logical and, `k and 1`.  In the other fragmentation function, you'll find something similar: `r = k%2`, which is equivalent to `r = k&1`.  It's my expectation that `k&1` should be faster in an interpreted language like python, because divisions (and therefore modulo) are slow (and Python isn't smart enough to do such code transformations -- different story in C, where I'd prefer the more-readable modulo).  Some benchmarks:

Python 3.6.7:
    In [1]: from random import randint
        ...: R = [randint(0, 2**64) for _ in range(1000000)]
        ...: %timeit z = sum(x&1 for x in R)
        ...: %timeit z = sum(x%2 for x in R)
        ...: 
    10 loops, best of 3: 45.5 ms per loop
    10 loops, best of 3: 71.5 ms per loop

Python 2.7.15rc1:
    In [1]: from random import randint
       ...: R = [randint(0, 2**64) for _ in range(1000000)]
       ...: %timeit z = sum(x&1 for x in R)
       ...: %timeit z = sum(x%2 for x in R)
       ...: 
    10 loops, best of 3: 70.7 ms per loop
    10 loops, best of 3: 89.7 ms per loop

Reply:
OH! I see. You're just grabbing the last bit. Thanks for the clarification :)